src/chart/generateCategoricalChart.tsx

```diff
@@ -1516,11 +1518,26 @@ export const generateCategoricalChart = ({
           ),
         });
       } else if (!_.isNil(data.activeTooltipIndex)) {
-        const { chartX, chartY, activeTooltipIndex } = data;
+        const { chartX, chartY } = data;
+        let { activeTooltipIndex } = data;
         const { offset, tooltipTicks } = this.state;
         if (!offset) {
           return;
         }
+        if (typeof syncMethod === 'function') {
+          // Call a callback function. If there is an application specific algorithm
+          activeTooltipIndex = syncMethod(activeTooltipIndex, data);
```

Comment:
Should this be `activeTooltipIndex = syncMethod(tooltipTicks, data);`?

`activeTooltipIndex` is already in the `data` object and without `tooltipTicks`, we can't do much with `syncMethod` function.

Reply:
You are right. Sorry I missed this.
I'm just raising a new PR to fix this

Reply:
No worries, I missed it too. Thank you!

Reply:
https://github.com/recharts/recharts/pull/2466 raised
If you find any other issues, please let me know. Thanks!